src/QueryStrategies/BatchQueryStrategies/batch_qs_base.jl

```diff
@@ -0,0 +1,130 @@
+    all_history_values = (length(history) > 0) && (collect(Iterators.flatten(history)))
+    candidate_indices = [i for i in pool_map[:U] if global_indices[i] ∉ all_history_values]
+    debug(getlogger(@__MODULE__), "[QS] Selecting best batch of $(qs.k) from $(length(candidate_indices)) candidates.")
+    local_query_indices = select_batch(qs, query_data, pool_map, candidate_indices)
+    return global_indices[local_query_indices]
+end
+
+"""
+rep_measure computes representativeness of observations.
+Only KDE is supported so far.
+"""
+function get_rep_measure(name::Symbol)::Function
+    if (name == :KDE)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, labels::Dict{Symbol, Vector{Int}}, candidate_indices::Vector{Int}) -> begin
+            γ = MLKernels.getvalue(model.kernel_fct.alpha)
+            return multi_kde(data, γ)(data[:, candidate_indices])
+        end
+    else
+        return throw(ArgumentError("Invalid representativeness measure $(name) specified."))
+    end
+end
+
+"""
+div_measure computes diversity of all samples to one specific sample
+Iterative computation: only value for added sample needs to be computed, old_scores saves aggregated result
+"""
+function get_iterative_div_measure(name::Symbol)::Function
+    if (name == :AngleDiversity)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, candidate_indices::Vector{Int}, j::Int, old_scores::Vector{Float64}) -> begin
+            if model.data == data
+                # reuse model kernel matrix
+                K = SVDD.is_K_adjusted(model) ? model.K_adjusted : model.K
+            else
+                # compute a new kernel matrix because query data differs from model data
+                K = MLKernels.kernelmatrix(Val(:col), model.kernel_fct, data)
+            end
+            div_scores = [-abs(K[i,j]) / (sqrt(K[i,i]) * sqrt(K[j,j])) for i in candidate_indices]
+            if (length(old_scores) > 0)
+                div_scores = min.(div_scores, old_scores)
+            end
+            return div_scores
+        end
+    elseif (name == :EuclideanDistance)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, candidate_indices::Vector{Int}, j::Int, old_scores::Vector{Float64}) -> begin
+            # vector containing norms of columns
+            # use vec() as otherwise a row vector is returned as opposed to a column vector
+            div_scores = Distances.colwise(Distances.Euclidean(), data[:,candidate_indices], data[:,j])
+            if (length(old_scores) > 0)
+                div_scores = min.(div_scores, old_scores)
+            end
+            return div_scores
+        end
+    else
+        return throw(ArgumentError("Invalid diversity measure $(name) specified."))
+    end
+end
+
+"""
+div_measure computes diversity of the samples in a batch
+enumerative computation: compute diversity for the whole batch each time
```

Comment:
What does "each time" mean here?

Reply:
The BatchQs triggers the diversity computation. The calculation is done on the full batch from scratch which is different from the iterative computation `get_iterative_div_measure`.

The name is confusing though and I have renamed the function to `get_div_measure`.